filament/backend/src/vulkan/VulkanHandles.h

```diff
@@ -294,8 +309,16 @@ struct VulkanProgram : public HwProgram, fvkmemory::Resource {
         PushConstantDescription pushConstantDescription;
     };
 
+    struct PushConstantInfo {
+        VkCommandBuffer cmdbuf;
+        backend::ShaderStage stage;
+        uint8_t index;
+        backend::PushConstantVariant value;
+    };
+
     PipelineInfo* mInfo;
     VkDevice mDevice = VK_NULL_HANDLE;
+    std::vector<PushConstantInfo> mQueuedPushConstants;
```

Comment:
should we do a "reasonable" .reserve() for this, so we avoid too many heap allocations?